crowbar_framework/app/models/api/node.rb

```diff
@@ -131,7 +140,10 @@ def upgrade
         return false
       end
 
-      reboot_and_wait
+      unless reboot_and_wait
+        save_error_state("Error while rebooting the node")
+        return false
```

Comment:
the return can be dropped if we merge #960 as `save_error_state` will raise an exception and not reach that point ever